### ISSUE TITLE
"Implement this" error from typeof(this) on primitive types

### ISSUE THREAD

**Issue by philrz** [brimdata/brim#2656]
Repro is with Brim commit 1e0a608 using the attached test data [values.zson.gz](https://github.com/brimdata/brim/files/10667358/values.zson.gz).

As shown in the video below, after loading the data and executing the query `typeof(this)`, a stack dump occurs.

https://user-images.githubusercontent.com/5934157/217038220-4cb89c6d-5f28-4bbf-abc0-2183cab5fdb4.mp4

I isolated a few of the individual values and found I could repro with just the values of types `ip`, `net`, `time`, etc., so it looks like several are not yet covered.

Here's the full stack dump text:

```
Error: Implement this: {"value":{"kind":"primitive","name":"ip"},"type":{"kind":"primitive","name":"type"}}
    at createData (/Users/phil/work/brim/packages/zealot/dist/cjs/test/factory.js:53:11)
    at createField (/Users/phil/work/brim/packages/zealot/dist/cjs/test/factory.js:26:39)
    at Object.createRecord (/Users/phil/work/brim/packages/zealot/dist/cjs/test/factory.js:14:21)
    at /Users/phil/work/brim/dist/panes/results-pane/table-controller.js:74:53
    at Array.map (<anonymous>)
    at /Users/phil/work/brim/dist/panes/results-pane/table-controller.js:74:27
    at mountMemo (/Users/phil/work/brim/node_modules/react-dom/cjs/react-dom.development.js:17057:19)
    at Object.useMemo (/Users/phil/work/brim/node_modules/react-dom/cjs/react-dom.development.js:17517:16)
    at Object.useMemo (/Users/phil/work/brim/node_modules/react/cjs/react.development.js:1645:21)
    at Object.useTableValues (/Users/phil/work/brim/dist/panes/results-pane/table-controller.js:70:24)
```

**Comment by philrz** [comment]
Here's another find that was reported after filing that looks like it's already fixed in #2657. Note how the Inspector view lacks the `(int8)` decorators.

```
$ zq -Z int8-tarray.zson 
[
    1 (int8),
    2 (int8)
]
```

<img width="1250" alt="image" src="https://user-images.githubusercontent.com/5934157/217123004-a45aa0f2-f65a-4594-8d47-570911bb6a61.png">

**Comment by philrz** [comment]
Verified in Zui commit 3d7b387 using the attached test data [values-plus-int8-array.zson.gz](https://github.com/brimdata/brim/files/10678716/values-plus-int8-array.zson.gz) that consists of the original test data along with the array of `int8` values shown in the last comment above.

As shown in the video below, now the `typeof(this)` query output shows everything we'd expect and no stack dump, and the `(int8)` decorators appear on the elements of our `int8` array as we'd expect.

https://user-images.githubusercontent.com/5934157/217345919-69349d64-354a-439d-b907-24748a7d6c2e.mp4

Thanks @jameskerr!
